### PR TITLE
Fixed crash on being with another player proxying `NSURLSession`/`URLSession`

### PR DESCRIPTION
**Pull request by ArielDemarco**
# Overview
This PR addresses an issue that arises when our SDK is initialized either before or after other SDKs that proxy `NSURLSession` / `URLSession` (specifically, those that replace or wrap the delegate provided upon calling NSURLSession.init).

The goal is to detect if this substitution has occurred in order to avoid forwarding calls to this proxy, thus preventing infinite recursion.

### CODE REVIEW

File: Tests/EmbraceCoreTests/TestSupport/TestDoubles/Dummy.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
+//
```
Comment (atreat) [inline Tests/EmbraceCoreTests/TestSupport/TestDoubles/Dummy.swift:1]: Can we change this filename to match `DummyURLSessionInitWithDelegateSwizzler`

File: Sources/EmbraceCommonInternal/Protocols/SwizzleCache.swift [hunk 2]
```diff
@@ -16,22 +16,29 @@ public class SwizzleCache {
     struct OriginalMethod: Hashable {
         let method: Method
         let baseClass: AnyClass
+        let swizzlerClass: String
```
Comment (atreat) [inline Sources/EmbraceCommonInternal/Protocols/SwizzleCache.swift:19]: Why is this change necessary? Can you document what this String means?

File: Tests/EmbraceCoreTests/TestSupport/TestDoubles/Dummy.swift (hunk 1 above)
Comment (ArielDemarco) [reply Tests/EmbraceCoreTests/TestSupport/TestDoubles/Dummy.swift:1]: Oh yes! Dunno why It was saved as dummy 🤔  so weird!
Gonna change it 👍

File: Sources/EmbraceCommonInternal/Protocols/SwizzleCache.swift (hunk 2 above)
Comment (ArielDemarco) [reply Sources/EmbraceCommonInternal/Protocols/SwizzleCache.swift:19]: Sure, I can.
The change was needed because the cache is just saving the "BaseClass" (aka `NSURLSession`) and the method that we get from selector. To prevent any kind of problems when multiple classes swizzle the same selector, im also adding the name of the swizzler (for example, `DummyURLSessionInitWithDelegateSwizzler` or `URLSessionInitWithDelegateSwizzler`).

**Review by NachoEmbrace — APPROVED** [review]

**Review by atreat — APPROVED** [review]
